src/day7.scala

```diff
@@ -0,0 +1,89 @@
+// using scala 3.1.0
+
+package day7
+
+import scala.util.Using
+import scala.io.Source
+import scala.annotation.tailrec
+
+@main def part1(): Unit =
+  println(s"The solution is ${part1(readInput())}")
+
+@main def part2(): Unit =
+  println(s"The solution is ${part2(readInput())}")
+
+def readInput(): String =
+  Using.resource(Source.fromFile("input/day7"))(_.mkString)
+
+sealed trait Crabmarine:
+  def moveForward(): Crabmarine
+  def moveBackward(): Crabmarine
+  def horizontal: Int
+  def fuelCost: Int
+
+case class ConstantCostCrabmarine(horizontal: Int) extends Crabmarine:
+  def fuelCost: Int = 1
+  def moveForward(): Crabmarine = this.copy(horizontal = horizontal + 1)
+  def moveBackward(): Crabmarine = this.copy(horizontal = horizontal - 1)
+
+case class IncreasingCostCrabmarine(horizontal: Int, fuelCost: Int = 1)
+    extends Crabmarine:
+  def moveForward() =
+    this.copy(horizontal = horizontal + 1, fuelCost = fuelCost + 1)
+  def moveBackward() =
+    this.copy(horizontal = horizontal - 1, fuelCost = fuelCost + 1)
+
+case class Crabmada(crabmarines: List[Crabmarine]):
+
+  require(crabmarines.nonEmpty)
+
+  @tailrec
+  final def align(
+      situation: List[Crabmarine] = crabmarines,
+      fuelCost: Int = 0
+  ): Int =
+    val allTheSame = situation.forall(_.horizontal == situation.head.horizontal)
+    if allTheSame then fuelCost
+    else
+      val maxHorizontal = situation.maxBy(_.horizontal)
+      val minHorizontal = situation.minBy(_.horizontal)
+
+      val fuelCostForTop = situation.collect {
```

Comment:
```suggestion
      val fuelCostForMax = situation.collect {
```
to be consistent everywhere.

Reply:
Fixed!